RFunction.R

```diff
@@ -436,31 +416,67 @@ rFunction = function(data, travelcut,
   )
   
   # Log results
-  logger.trace(paste0("   ", sum(data$behav == "SResting", na.rm = T), " locations classified as SResting"))
-  logger.trace(paste0("   ", sum(data$behav == "STravelling", na.rm = T), " locations classified as STravelling"))
+  logger.info(paste0("   |> ", sum(data$behav == "SResting", na.rm = T), " locations classified as SResting"))
+  logger.info(paste0("   |> ", sum(data$behav == "STravelling", na.rm = T), " locations classified as STravelling"))
   
   
-  #### 2. Altitude Classification ----
-  logger.info("[2] Performing altitude classification")
-  data %<>%
-    mutate(
-      RULE = case_when(
-        (behav == "SResting") & (altchange == "ascent") ~ "[2] Altitude increasing",
-        (behav == "SResting") & (altchange == "descent") & (lead(altchange) %in% c("descent", "ascent")) ~ "[2] Altitude decreasing",
-        TRUE ~ RULE
-      ),
-      behav = case_when(
-        (behav == "SResting") & (altchange == "ascent") ~ "STravelling",
-        (behav == "SResting") & (altchange == "descent") & (lead(altchange) %in% c("descent", "ascent")) ~ "STravelling",
-        TRUE ~ behav
+  ## [2] Altitude Classification ----
+  
+  #' Remaining resting locations reclassified as travelling according to the following rules:
+  #' (i) If a bird is ascending ==> STravelling
+  #' (ii) If a bird is descending AND:
+  #'      Next location is ascending/descending ==> STravelling
+  #'      Next location is flatlining ==> remains SResting
+  #' (iii) If a bird is flatlining, it remains SResting
+  #' 
+  #' NOTE: stationary locations are not re-classified here, so this step can
+  #' produce locations with both `stationary == 1` AND `behav == STravelling`.
+  #' This can be a bit counter-intuitive, but we don't want to change it,
+  #' otherwise the roosting-site identification performed in Data Prep would be
+  #' outdated
+  
+  if(alt_classify){
+  
+    logger.info("[2] Performing altitude classification")
+    
+    data %<>%
+      # QUESTION (BC): shouldn't this step be grouped by bird given we're using `lead()`?
```

Comment:
yes this seems sensible

Reply:
This is briefly expended in issue #11. Basically, it won't make a difference in the output since `altchange` was derived by ID